whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticFullTextIndex.java

```diff
@@ -67,6 +68,38 @@ public void scheduledUpdate() {
         LOGGER.info("Completed updating Elasticsearch indexes");
     }
 
+    @Scheduled(fixedDelay = 60 * 60 * 1_000L)
+    @SchedulerLock(name = ALERT_TASK_NAME)
+    public void checkIndex() {
+        if (!elasticIndexService.isEnabled()) {
+            LOGGER.error("Elasticsearch is not enabled");
+            return;
+        }
+
+        try {
+            alert();
+        } catch (DataAccessException | IOException | IllegalStateException e) {
+            LOGGER.error("Unable to perform fulltext index checks due to {}: {}", e.getClass(), e.getMessage());
+        }
+
+        LOGGER.info("Completed Elasticsearch index check");
+    }
+
+    protected void alert() throws IOException{
+        if (shouldRebuild()) {
+            LOGGER.error("ES indexes needs to be rebuild");
+            return;
+        }
+
+        final Map<Integer, Integer> maxSerialIdWithObjectCount = serialDao.getMaxSerialIdWithObjectCount();
+        final int countInDb = ((int) maxSerialIdWithObjectCount.values().toArray()[0]);
```

Comment:
This will not work. As current serial Id in database might be behind in ES index. You need to get number of objects for a serialId saved in ES and then compare otherwise this will always result in error

Reply:
Updated